spec/models/spree/order/payment_spec.rb

```diff
@@ -4,18 +4,46 @@
 
 module Spree
   describe Spree::Order do
+    before { Stripe.api_key = "sk_test_12345" }
     let(:order) { build(:order) }
     let(:updater) { OrderManagement::Order::Updater.new(order) }
-    let(:bogus) { create(:bogus_payment_method, distributors: [create(:enterprise)]) }
+    let(:payment_method) {
+      create(:stripe_sca_payment_method, distributor_ids: [create(:distributor_enterprise).id],
+                                         preferred_enterprise_id: create(:enterprise).id)
+    }
+    let(:source) {
+      create(:credit_card)
+    }
+    let(:payment1) {
+      create(:payment, amount: 50, payment_method:, source:, response_code: "12345")
+    }
+    let(:payment2) {
+      create(:payment, amount: 50, payment_method:, source:, response_code: "12345")
+    }
+    let(:payment3) {
+      create(:payment, amount: 50, payment_method:, source:, response_code: "12345")
+    }
+    let(:failed_payment) {
+      create(:payment, amount: 50, state: 'failed', payment_method:, source:,
+                       response_code: "12345")
+    }
+    let(:payment_authorised) {
+      payment_intent(50, "requires_capture")
+    }
+    let(:capture_successful) {
+      payment_intent(50, "succeeded")
+    }
 
     before do
       allow(order).to receive_message_chain(:line_items, :empty?).and_return(false)
       allow(order).to receive_messages total: 100
+      stub_request(:get, "https://api.stripe.com/v1/payment_intents/12345").
+        to_return(status: 200, body: payment_authorised)
+      stub_request(:post, "https://api.stripe.com/v1/payment_intents/12345/capture").
+        to_return(status: 200, body: capture_successful)
```

Comment:
I noticed this was how a lot of the calls were mocked elsewhere in the codebase, so I just copy-pasted. I figured it was okay since this was the payment spec and not the stripe spec. We do have deeper coverage than before with the Bogus gateway, since now it actually validates the payment's state before sending it on. That's why I was having issues with it before.